File: app/src/main/java/com/app/turingrobot/core/dagger/component/AppComponent.java
package com.app.turingrobot.core.dagger.component;


import com.app.turingrobot.core.App;
import com.app.turingrobot.core.RobotService;
import com.app.turingrobot.core.dagger.module.AppModule;
import com.app.turingrobot.helper.SpfHelper;
import com.google.gson.Gson;

import javax.inject.Singleton;

import dagger.Component;
import okhttp3.OkHttpClient;

/*
 * Created by sword on 2017/3/27.
 */
@Singleton
@Component(modules = AppModule.class)
public interface AppComponent {

    Gson getGson();

    SpfHelper getSpfHelper();

    App getApp();

    OkHttpClient getOkhttp();

    RobotService getService();

}

File: app/src/main/java/com/app/turingrobot/core/dagger/component/AppComponent.kt
package com.app.turingrobot.core.dagger.component


import com.app.turingrobot.core.dagger.module.AppModule
import com.app.turingrobot.ui.core.BaseActivity
import com.app.turingrobot.ui.core.BaseFragment

import javax.inject.Singleton

import dagger.Component

/*
 * Created by sword on 2017/3/27.
 */
@Singleton
@Component(modules = arrayOf(AppModule::class))
interface AppComponent {

    fun inject(activity: BaseActivity)

    fun inject(fragment: BaseFragment)

}
